src/provider/provider_os_memory_linux.c

```diff
@@ -28,6 +31,21 @@ umf_result_t os_translate_mem_visibility_flag(umf_memory_visibility_t in_flag,
     return UMF_RESULT_ERROR_INVALID_ARGUMENT;
 }
 
+// open or create a shared memory file
+int os_shm_open(const char *shm_name) {
+    if (shm_name == NULL) {
+        LOG_ERR("empty name of a shared memory file");
+        return -1;
+    }
+
+    int fd = shm_open(shm_name, O_RDWR, 0600);
```

Comment:
Could you try passing the O_RDWR|O_CREAT|O_EXCL flags? Maybe that would help with the initial ftruncate?

Reply:
Yes, With the `O_EXCL` flag it does work at last :-)

I added an option to create an SHM file if open() fails.

Done.